packages/radix-ui-themes/changelog.md

```diff
@@ -5,6 +5,9 @@
 - General
   - Improve ESM compatibility
   - Improve tree-shaking of individual component parts
+  - [**Breaking**] Remove component prop definitions and internal helpers from the root `@radix-ui/themes` import entry point and export them from `@radix-ui/themes/props` and `@radix-ui/themes/helpers` to make it possible to build your own component library on top of Radix Themes using the same techniques.
+    - Note: you might need to use `"moduleResolution": "NodeNext"` with your compiler to access these paths
+    - Note: changes to prop defs and helpers won’t be covered by semver
```

Comment:
I wonder if technically changes to prop defs are covered by semver as they are linked to changes in the component's APIs anyway? Or I guess you meant that if we somehow change the structured way those are exposed (not changing the props themselves) then we don't cover these changes via semver?

Reply:
Yeah, like if we decide to remove/rework a prop def, e.g. something odd like `inheritedColorProp`, this shouldn't be covered by semver. Semver would of course still cover changes to the public component API, but the internal prop def changes should be up to us until this area is more mature and until we know about more similar use cases with a design system that extends Radix Themes.